onnx_caffe2/backend.py

```diff
@@ -365,16 +376,25 @@ def _create_rnn(cls, init_model, pred_model, n, opset_version):
         activation = attrs.pop('activations')[0]
         assert not attrs, "unsupported RNN attributes: " + str(attrs.keys())
 
-        input_blob, W, R, B, sequence_lens, initial_h = n.inputs
+        input_blob, W, R, B, _, initial_h = n.inputs
 
-        input_size = cls._rnn_shape_inference(init_model, pred_model, n, input_blob, W)
-        if input_size is None:
+        try:
+            seq_length, batch_size, input_size = cls._rnn_shape_inference(init_model, pred_model, n, input_blob, W)
+        except:
```

Comment:
You don't need/want plain `except` here (e.g. `KeyboardInterrupt`, in which case you definitely don't want to tell users it's shape inference failed). At least `except Exception`.